src/main/java/teammates/ui/controller/Action.java

```diff
@@ -224,12 +224,12 @@ private boolean isNotLegacyLink() {
     }
 
     private boolean doesUserNeedToLogin(UserType currentUser) {
-        boolean userNeedsGoogleAccountForPage =
+        boolean shouldUserNeedGoogleAccountForPage =
                 !Const.SystemParams.PAGES_ACCESSIBLE_WITHOUT_GOOGLE_LOGIN.contains(request.getRequestURI());
-        boolean userIsNotLoggedIn = currentUser == null;
-        boolean noRegkeyGiven = getRegkeyFromRequest() == null;
+        boolean isUserNotLoggedIn = currentUser == null;
```

Comment:
hmm.. alright then. There are also cases of `if (!isUserLoggedIn) { ... }` though